### PR TITLE
Serdes for custom list

### PR DESCRIPTION
**Pull request by qnkhuat**
In https://github.com/metabase/metabase/pull/27343, https://github.com/metabase/metabase/pull/27403 we added custom list for dashboard and card.

We did that by introducing 2 new keys on the parameter.
These parameters are stored in `card.parameters` and `dashboard.parameters` as stringified JSON.

```
{:name                 "Static Category",
 :slug                 "static_category"
 :id                   "_STATIC_CATEGORY_",
 :type                 "category",
+ :values_source_type   "card"
+ :values_source_config {:card_id     source-card-id
+                        :value_field [:field 3 nil]}}
```

This PR makes sure serdes works nicely with `card_id` and `value_field` in the parameter config.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27422)

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj [hunk 1]
```diff
@@ -240,108 +240,108 @@
...
-                  (is (= (clean-entity timeline)
-                         (->> (db/select-one 'Timeline :entity_id entity_id)
-                              (serdes.base/extract-one "Timeline" {})
-                              clean-entity)))))
-
-              (testing "for settings"
-                (is (= (into {} (for [{:keys [key value]} (get @entities "Setting")]
-                                  [key value]))
-                       (yaml/from-file (io/file dump-dir "settings.yaml")))))))))))
+        (testing "ingest and load"
```
Comment (qnkhuat) [inline enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj:243]: no logical changes here. just indent.

File: src/metabase/models/serialization/base.clj [hunk 2]
```diff
@@ -258,6 +258,39 @@
 (defmethod extract-one :default [model-name _opts entity]
   (extract-one-basics model-name entity))
 
+(defmulti serdes-descendants
```
Comment (qnkhuat) [inline src/metabase/models/serialization/base.clj:261]: moved this up from under because this is part of the serialization process.

Comment (bshepherdson) [reply src/metabase/models/serialization/base.clj:262]: Fair. `serdes-descendants` was added for selective serialization late in the process, so I guess it landed low in the file.

File: src/metabase/models/serialization/util.clj [hunk 3]
```diff
@@ -405,6 +404,31 @@
...
+  "Given the :parameter field of a `Card` or `Dashboard`, as a vector of maps, converts
+  it to a portable form with the CardIds/FieldIds replaced with `[db schema table field]` references."
+  [parameters]
+  (map ids->fully-qualified-names parameters))
+
+(defn import-parameters
+  "Given the :parameter field as exported by serialization convert its field references
+  (`[db schema table field]`) back into raw IDs."
+  [parameters]
+  (->> parameters
```
Comment (bshepherdson) [inline src/metabase/models/serialization/util.clj:417]: This seems more legible as `(for [param parameters] (-> param mbql-fully.qualified-names->ids (m/update-existing-in ...)))`.

File: enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj [hunk 4]
```diff
@@ -940,3 +939,97 @@
                        :template-tags
                        (get "snippet: things")
                        :snippet-id)))))))))
+
+(deftest card-has-parameter-with-source-card-test
```
Comment (bshepherdson) [inline enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj:943]: There's no tests I can see for correct serdes with a literal value list. Is that a separate PR? I don't see any reason why it shouldn't Just Work, but we should test it.

**Review by bshepherdson — APPROVED** [review]

File: enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj (hunk 4 above)
Comment (bshepherdson) [reply enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj:943]: Also consider adding this new kind of linking to the test data generation used by the e2e test.

That system is kind of weird, I'm happy to point the way, or pair on it.

**Pull request merged**